hooks/useTokenPayment.js

```diff
@@ -9,104 +9,85 @@ import useSWR from 'swr';
 import {
   BLOCKCHAIN_DAO_DIAMOND_ADDRESS,
   BLOCKCHAIN_DAO_TOKEN,
-  BLOCKCHAIN_DAO_TOKEN_ABI,
   BLOCKCHAIN_DIAMOND_ABI,
 } from '../config_blockchain';
 import { fetcher } from '../utils/blockchain';
 
 dayjs.extend(dayOfYear);
 
-export const useTokenPayment = ({ value, startDate, totalNights }) => {
+export const useTokenPayment = ({ startDate, totalNights, dailyValue }) => {
   const { account, library } = useWeb3React();
   const [pendingTransactions, setPendingTransactions] = useState([]); //In general the following pendingTransactions state should be moved to the root of the app, and should be used as a dependency by all hooks that read blockchain state
-  const [isPending, setPending] = useState(false); //Used when need to make several blockchain transactions in a row
-
-  const start = dayjs(startDate);
-  const bookingYear = start.year();
-  let nights = [[bookingYear, dayjs(startDate).dayOfYear()]];
-  for (var i = 1; i < totalNights; i++) {
-    nights = [
-      ...nights,
-      [bookingYear, dayjs(startDate).add(i, 'day').dayOfYear()],
-    ];
-  }
-
-  const { data: balanceLocked } = useSWR(
-    [BLOCKCHAIN_DAO_DIAMOND_ADDRESS, 'lockedStake', account],
+  const [isPending, setPending] = useState(false);
+  const bookingYear = dayjs(startDate).year();
+  const totalNightsContractForm = Array.from(
+    { length: totalNights },
+    (_, i) => [bookingYear, dayjs(startDate).add(i, 'day').dayOfYear()],
+  );
+  const { data: bookedNights, mutate: updateBookedNights } = useSWR(
+    [
+      BLOCKCHAIN_DAO_DIAMOND_ADDRESS,
+      'getAccommodationBookings',
+      account,
+      bookingYear,
+    ],
     {
       fetcher: fetcher(library, BLOCKCHAIN_DIAMOND_ABI),
     },
   );
 
-  const tokensToStake =
-    balanceLocked &&
-    BigNumber.from(value)
-      .mul(BigNumber.from(10).pow(BLOCKCHAIN_DAO_TOKEN.decimals))
-      .sub(balanceLocked);
+  const isStakingConfirmedOnBlockchain = useMemo(() => {
+    if (!bookedNights) return false;
+    const areAllNightsBooked = totalNightsContractForm.every((night) => {
+      const [year, dayOfYear] = night;
+      return bookedNights.some(
+        (bookedNight) =>
+          bookedNight.year === year && bookedNight.dayOfYear === dayOfYear,
+      );
```

Comment:
Let us also check here if the number of tokens staked per night is the right one: 

formatBigNumberForDisplay(bookedNights[i].price,BLOCKCHAIN_DAO_TOKEN.decimals,0)

Reply:
ok, though I am not sure if I understand what `formatBigNumberForDisplay` is returning. I see it will return `null` if `bookedNights[i].price` is not a `BigNumber`, but this is not what you want to check, correct?